src/scripts/run_certbot.sh

```diff
@@ -87,6 +114,14 @@ for conf_file in /etc/nginx/conf.d/*.conf*; do
             key_type="rsa"
         fi
 
+        # Determine the authenticator to use to solve the challenge
+        if [[ "${cert_name,,}" =~ ^.*(-|\.)(dns-($(echo ${CERTBOT_DNS_AUTHENTICATORS} | sed 's/ /|/g'))).*$ ]]; then
+            challenge_authenticator=${BASH_REMATCH[2]}
+            debug "Found mention of authenticator '${challenge_authenticator}' in name '${cert_name}'"
+        else
+            challenge_authenticator="${CERTBOT_DEFAULT_AUTHENTICATOR:-webroot}"
+        fi
+
         # Find all 'server_names' in this .conf file and assemble the list of
         # domains to be included in the request.
         domain_request=""
```

Comment:
DNS challenges _allow_ to support wildcard domains, but do not necessarily mean that you want a certificate for a wildcard domain. One of the big interests of DNS challenges is that you can get a certificate even if the traffic is not directly pointing at you at the moment (allowing for instance to have a local server with valid certificates that's not reachable from outside). At the moment, I'm rewriting my server's configuration to be in containers, and I want the setup fully working before migrating where the traffic points to: DNS challenges allow me to do that, webroot wouldn't. And if I want to support a part of my domains through wildcards, I can! But I don't _have to_ (and most of my domains won't have wildcard handling).

Let's Encrypt will by itself complain if you have a wildcard and a conflicting domain with it (e.g. `*.example.com` and `first.example.com` would conflict together in the cert) but will allow non-conflicting domains together (e.g. `*.example.com` and `*.first.example.com` will work, since the wildcard is supposed to cover only a single level). I believe the domains logic should be kept as it is right now, and people should fix their configuration accordingly when using wildcards. If you declare your server name entry with both `*.example.com` and `first.example.com` for instance, the latter is redundant, so that'd be good practice to just keep the wildcard anyway, I think!